gulpfile.js

```diff
@@ -85,13 +88,44 @@ function fetchS3(params, done) {
   downloader.on('end', done);
 }
 
-function pushS3(params, done) {
+function pushS3(params) {
   var uploader = s3Client.uploadFile(params);
   uploader.on('error', S3_CONSTS.handleError.bind(uploader));
   uploader.on('progress', S3_CONSTS.handleProgress.bind(uploader, 'Uploading', params.localFile));
-  uploader.on('end', done);
+  return Promise.fromCallback(function(cb) {
+    uploader.on('end', cb);
+  });
 }
 
+
+// Look for any troubled path lengths so we don't run into problems on the Windows builds: https://github.com/gitterHQ/desktop/issues/59
+// If you are running into issues, you can just install the sub-depedency on the root level
+// This shouldn't be a problem if we decide to require npm 3 for the desktop builds
+gulp.task('check-path-safety-for-windows', function() {
+  var stream = gulp.src('./nwapp/**/*', {read: false})
+    .pipe(through.obj(function(chunk, enc, cb) {
+        var baseWindowsCheckPath = 'C:\\Users\\some-longish-username\\AppData\\Local\\Temp\\nw2128_17940';
+        var pathToCheck = path.join(baseWindowsCheckPath, path.relative(chunk.base, chunk.path));
+
+        if(pathToCheck.length > 256) {
+            var nodeModulesMessage = '';
+            if(pathToCheck.match('node_modules')) {
+              nodeModulesMessage = ' --- You can try installing a sub-depedency as a root-level module to shorten up the paths';
+            }
+          throw new gutil.PluginError('checking-path-lengths', {
+            message: 'You have a path length that exceeds 256 characters and will cause issues on Windows: ' + chunk.path + '. Note, we checked with a base path: ' + baseWindowsCheckPath + nodeModulesMessage
+          });
+        }
+
+        this.push(chunk);
+        cb();
+    }));
+
+    // Avoid the high-water mark: https://github.com/gulpjs/gulp/issues/1356
+    return exhaustively(stream);
+});
```

Comment:
I'd rather leave the check as it is still possible that it could happen. When you get this failure, it isn't clear what is causing it so the pre-check just gives some peace of mind and catches a potential problem.

re: this can be removed because we use npm 3 to flatten dependencies.